### ISSUE TITLE
More than one incoming spans

### ISSUE THREAD

**Issue by TanYuzhen** [Sachin-A/TraceWeaver#1]
In the code that handles in_coming_spans in the executor.py file, you can see that TraceWeaver can't handle a microservice with multiple incoming spans. For example, in DeathStarBench's Media microservice's compose-review-service, it has multiple incoming spans, but it doesn't seem to handle this case in the actual code implementation, can you please tell me what is the solution for this problem?

**Comment by StLeoX** [comment]
确实。`assert len(in_span_partitions) == 1` 之类的逻辑，表明上游服务（端点）只能有一个。也许算法的适用场景是 DAG，而不是严格的 Tree 结构。也就意味着像 [compose-review-service](https://github.com/delimitrou/DeathStarBench/tree/master/mediaMicroservices) 这样入度大于一的服务依赖关系不被允许。

**Comment by StLeoX** [comment]
作者在[视频](https://www.youtube.com/watch?v=aQnrwPMURwA&t=11s)中给出了这个结论，但似乎并没有给出详细分析。

**Comment by TanYuzhen** [comment]
在视频哪个位置哪，顺便问一下你也是WHU的吗

**Comment by StLeoX** [comment]
> 在视频哪个位置哪
3min 左右
![Image](https://github.com/user-attachments/assets/4d2c11c9-0e10-4e8d-897e-a97cf13515d0)

**Comment by StLeoX** [comment]
> 顺便问一下你也是WHU的吗
